proxy_agent_setup/src/main.rs

```diff
@@ -24,13 +24,25 @@ const SERVICE_DISPLAY_NAME: &str = "Microsoft Azure Guest Proxy Agent";
 const SERVICE_NAME: &str = "azure-proxy-agent";
 
 fn main() {
-    logger::init_logger();
+    let format = tracing_subscriber::fmt::format()
+        .with_level(true)
+        .with_thread_names(true);
+    // Configurable via environment variable for now
+    // https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/filter/struct.EnvFilter.html#directives
+    let stderr_layer = tracing_subscriber::fmt::layer()
+        .event_format(format)
+        .with_writer(std::io::stderr)
+        .with_filter(tracing_subscriber::EnvFilter::from_env(
+            "GUEST_PROXY_AGENT_LOG",
+        ));
+    let registry = tracing_subscriber::registry().with(stderr_layer);
+    tracing::subscriber::set_global_default(registry).expect("Unable to configure logging!");
```

Comment:
I'm hesitant to panic here. If the issue is permanent it would result in the VM losing the ability to communicate. It will restart but will loop.

Reply:
According to tracing's [documentation](https://docs.rs/tracing/0.1.40/tracing/subscriber/fn.set_global_default.html) it only panics if the global default has already been set, which is either programmer error or a library that is misbehaving. In either case I would hope that it is caught by CI before being published.

However, there is a small chance that in the future the API changes and it can fail for some other reason, so this approach does have downsides. If we do stick with a panic here, I do think I should write a clearer message explaining it's due to programmer error.

How would you recommend proceeding if we can't configure a logger and don't panic? I guess we can announce the failure on stderr, but that's also less noticeable to an administrator than a service failing.